src/ElementComparator.h

```diff
@@ -0,0 +1,84 @@
+//
+//  ElementComparator.h
+//  drafter
+//
+//  Created by Jiri Kratochvil on 28/02/18.
+//  Copyright (c) 2018 Apiary Inc. All rights reserved.
+//
+#ifndef DRAFTER_ELEMENTCOMPARATOR_H
+#define DRAFTER_ELEMENTCOMPARATOR_H
+
+#include "refract/InfoElements.h"
+#include <set>
+#include <algorithm>
+
+namespace drafter
+{
+    using namespace refract;
+    struct InfoElementsComparator {
+        bool operator()(const InfoElements& rhs, const InfoElements& lhs)
+        {
+
+            const std::set<std::string> ignore = { "sourceMap" };
+            std::set<std::string> keys;
+
+            if (std::find_if(lhs.begin(),
+                    lhs.end(),
+                    [&keys, &rhs, &ignore](const auto& info) {
+                        auto key = info.first;
+                        if (ignore.find(key) != ignore.end()) {
+                            return false;
+                        }
+                        keys.insert(key);
+
+                        auto rInfo = rhs.find(key);
+                        if (rInfo == rhs.end()) {
+                            return true;
+                        }
+
+                        return !(*rInfo->second == *info.second);
+                    })
+                != lhs.end()) {
+                return false;
+            }
+
+            if (std::find_if(rhs.begin(),
+                    rhs.end(),
+                    [&keys, &lhs, &ignore](const auto& info) {
+                        auto key = info.first;
+                        if (ignore.find(key) != ignore.end()) {
+                            return false;
+                        }
+
+                        if (keys.find(key) != ignore.end()) {
+                            return false;
+                        }
+
+                        return true;
+                    })
+                != rhs.end()) {
+                return false;
```

Comment:
Same as above. `gcov` excerpt:
```
     2448:   45:            if (std::find_if(rhs.begin(),
        -:   46:                    rhs.end(),
      846:   47:                    [&keys, &lhs, &ignore](const auto& info) {
      560:   48:                        auto key = info.first;
      560:   49:                        if (ignore.find(key) != ignore.end()) {
      277:   50:                            return false;
        -:   51:                        }
        -:   52:
        6:   53:                        if (keys.find(key) != ignore.end()) {
        3:   54:                            return false;
        -:   55:                        }
        -:   56:
    #####:   57:                        return true;
      816:   58:                    })
     2448:   59:                != rhs.end()) {
    #####:   60:                return false;
        -:   61:            }

```

Reply:
Non relevant according to change algorihm